### PR TITLE
Add support for showing geometries in Map component

### PR DESCRIPTION
**Pull request by bjosttveit**
## Description

**Pre-release version**: `v4.8.0-geometrymap.preview2`

This PR adds support for showing geometries in either `GeoJSON` or `WKT` format in the Map component.

Geometries are read from the data model, by pointing to an array in `dataModelBindings` like this:
```json5
...
"dataModelBindings": {
  "geometries": "MapData.Geometries"
},
...
```

The array needs to be defined in the data model as follows, having the properties `data`, and `label` as strings:

```json5
...
"type": "array",
"items": {
  "type": "object",
  "properties": {
    "data": { "type": "string" },
    "label": { "type": "string" }
  }
}
...
```

The `data` field can contain either `WKT` or stringified `GeoJSON`. The component tries to parse as `GeoJSON` by default, but can be changed by setting the following property on the Map component:
```json5
...
"geometryType": "WKT",
...
```

The component will center and zoom the map to fit the geometries. This will override any `centerLocation` and `zoom` set on the component.

### Example configuration

```json5
{
  "id": "map",
  "type": "Map",
  "textResourceBindings": {
    "title": "Områder",
  },
  "dataModelBindings": {
    "geometries": "MapData.Geometries"
  },
  "geometryType": "WKT",
},
```

bilde

Note that the tooltips with the label are always visible, this is because the geometries become almost invisible when zooming out far enough.

### Other changes
- Moved map implementation from `altinn-design-system` to this repo
- Made `simpleBinding` (and setting a marker) optional
- Made marker draggable
- Map is no longer faded in PDF, only in regular summary
- Attribution (copyright notice) is styled better for PDF
- Set minimum zoom and max bounds so it looks less buggy when moving around
- Added print media emulation and snapshots for PDF tests, so we can actually catch mistakes with the PDF view. The map marker was at one point missing from the PDF because it had `role=button`, and we hide buttons by default in PDF, things like this should now be caught by percy. Also setting the viewport to simulate A4 paper size before taking the snapshots but after #readyForPrint is visible. This way we simulate how the PDF-generator works as well.
  - **Edit**: It seems like percy does not actually take a screenshot but rather a snapshot of the dom and renders that in its own service, so the print emulation is not actually active in the snapshots.
- Added cypress tests for old and new functionality of the map, and also added map to existing PDF tests.

## Related Issue(s)

- closes https://github.com/Altinn/altinn-studio/issues/12984

## Verification/QA

- Manual functionality testing
  - [x] I have tested these changes manually
  - [ ] Creator of the original issue (or service owner) has been contacted for manual testing (or will be contacted when released in alpha)
  - [ ] No testing done/necessary
- Automated tests
  - [x] Unit test(s) have been added/updated
  - [x] Cypress E2E test(s) have been added/updated
  - [ ] No automatic tests are needed here (no functional changes/additions)
  - [ ] I want someone to help me make some tests
- UU/WCAG ([follow these guidelines](https://aksel.nav.no/god-praksis/artikler/utvikling?tema=universell-utforming) until we have our own)
  - [ ] I have tested with a screen reader/keyboard navigation/automated wcag validator
  - [ ] No testing done/necessary (no DOM/visual changes)
  - [ ] I want someone to help me perform accessibility testing
- User documentation @ [altinn-studio-docs](https://github.com/Altinn/altinn-studio-docs)
  - [ ] Has been added/updated
  
  - [ ] No functionality has been changed/added, so no documentation is needed
  - [ ] I will do that later/have created an issue
  
- Support in Altinn Studio
  - [ ] Issue(s) created for support in Studio
  
  - [ ] This change/feature does not require any changes to Altinn Studio
- Sprint board
  - [x] The original issue (or this PR itself) has been added to the Team Apps project and to the current sprint board
  - [ ] I don't have permissions to do that, please help me out
- Labels
  - [x] I have added a `kind/*` label to this PR for proper release notes grouping
  - [ ] I don't have permissions to add labels, please help me out

### CODE REVIEW

**Comment by MartinRoberg** [comment]
Vi i DMF testa komponenten litt. Den fungerer generelt bra, spesielt som vanlig komponent.
Derimot fant vi et issue når det kommer til rendering av kartet som summary i pdf. 

Dette issuet ser vi ikke på som en blokker for denne pr-en, men beskriver det under uansett.

Vi bruker WKT og har valgt tre geometrier. (Uten zoom og centerlocation satt)
De tre geometriene vises da basert på bounds.

I browser vises disse da slik, hvor man kan se alle tre geometriene:
image

Derimot, når dette rendres i pdf-en så virker det som at den kutter kart-komponenten litt.
I pdf ser da kart-komponenten slik ut:
image


**Legger til data/config vi har brukt under:**

_Geometry data_
```js
[
  {
    "label": "Espe 3", 
    "data": "POLYGON ((9.55524330567219 61.4339474557278,9.58992047338494 61.4338016909585,9.58911658674369 61.3911615077072,9.55448666763505 61.3913070157442,9.55524330567219 61.4339474557278))"
  },
  {
    "label": "Espe 4", 
    "data": "POLYGON ((9.58992047338494 61.4338016909585,9.62459725824029 61.4336470988522,9.62374612469205 61.3910071878778,9.58911658674369 61.3911615077072,9.58992047338494 61.4338016909585))"
  },
  {
    "label": "Espe 5", 
    "data": "POLYGON ((9.62459725824029 61.4336470988522,9.65927363774107 61.4334836796041,9.65837525908263 61.3908440564506,9.62374612469205 61.3910071878778,9.62459725824029 61.4336470988522))"
  },
]
```

_Map component config for **web**_
```js
{
        "dataModelBindings": {
          "geometries": "Geometries"
        },
        "readOnly":  false,
        "geometryType": "WKT",
        "id": "ValgteBergrettKartVisning",
        "type": "Map"
}
```

_Map component config for **PDF**_
```js
{
        "dataModelBindings": {
          "geometries": "Geometries"
        },
        "readOnly": true,
        "renderAsSummary": true,
        "geometryType": "WKT",
        "id": "ValgteBergrettKartVisningPDF",
        "type": "Map"
}
```

File: src/layout/Map/utils.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+export function isLocationValid(location: Location | undefined): location is Location {
+  return typeof location?.latitude === 'number' && typeof location?.longitude === 'number';
+}
+
+/**
+ * Center & Zoom / Bounds controls the starting view of the map in order of priority
+ * 1. If a marker is set, center on that with zoom=16
+ * 2. If custom center and/or zoom is set, use that
+ * 3. If neither, but there are geometries present, use bounds to center on those
+ * 4. Else, use default center and zoom
```
Comment (olemartinorg) [inline src/layout/Map/utils.ts:65]: This order is outdated now? :thinking:

**Review by olemartinorg — APPROVED** [review]
Looks great! :raised_hands:

File: src/layout/Map/utils.ts (hunk 1 above)
Comment (bjosttveit) [reply src/layout/Map/utils.ts:65]: yes 🤦

**Pull request merged**